server/services/v1/query_runner.go

```diff
@@ -591,17 +580,31 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
 					UpdatedAt: row.Data.UpdatedToProtoTS(),
 				},
 			})
-			totalInPage++
 
 			if len(resp.Hits) == pageSize {
 				break
 			}
 		}
-		if len(resp.Hits) == 0 {
+		if rowReader.err != nil {
+			return nil, ctx, rowReader.err
```

Comment:
We may have buffered the events in the `Hits` if we return an error from here then we won't be able to send those hits?